Add cmd-d shortcut for (terminal) pane::SplitRight

jasikpark opened the pull request:
This matches the behavior of Visual Studio Code

Closes #ISSUE

Release Notes:

- Added VS Code's terminal split keybindings (`cmd-d` on MacOS, `ctrl-shift-5` on Windows and Linux)

jasikpark committed:
Add cmd-d shortcut for pane::SplitRight

This matches the behavior of Visual Studio Code

dinocosta commented:
> Not for the editor, but for the terminal. Though I'm not finding any docs that describe splitting the terminal in VS Code to provide alternative proof...

Got it! Sorry, I got confused and completely missed that this was terminal specific. I've been able to confirm that VS Code does indeed use `cmd-\` to split the editor and `cmd-d` to split the terminal, on MacOS. 

Since `cmd-d` in the `Terminal` context doesn't conflict with any existing binding let's go ahead and add this. Seeing as we're updating the MacOS ones, would you also be able to update the corresponding default bindings on Windows and Linux bindings to `ctrl-shift-5`? Thanks!

jasikpark committed:
Add split terminal shortcut for Linux and Windows

jasikpark commented:
Done!

dinocosta committed:
fix(keymaps): fix single colon

Update uses of `pane:SplitRight` to `pane::SplitRight` as having a
single colon would lead to this being an invalid binding.

jasikpark commented:
> Thank you @jasikpark ! I ended up pushing [a commit](https://github.com/zed-industries/zed/pull/41139/commits/0c6518be43e467da5de4e5017d10b1257e0f7d3d) to your branch and updating the release notes, will merge this now 🙂

Ah, the macOS keyboard shortcut is listed as just `cmd` rather than `cmd+D` in the release notes 😅 

Glad this got in though!

jasikpark commented:
cool, seems smart - I feel like I prefer the approach in https://github.com/changesets/changesets where the release notes are in a special file tracked for each PR <3

https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md

(rust version: https://github.com/knope-dev/knope)
